src/metabase/api/session.clj

```diff
@@ -231,7 +231,12 @@
         (let [reset-token        (user/set-password-reset-token! user-id)
               password-reset-url (str (public-settings/site-url) "/auth/reset_password/" reset-token)]
           (log/info password-reset-url)
-          (messages/send-password-reset-email! email nil password-reset-url is-active?))))))
+          (messages/send-password-reset-email! email nil password-reset-url is-active?)))
+      (events/publish-event! :event/password-reset-initiated
+                             {:user-id api/*current-user-id*
+                              ;; Fetch hashed value of reset token as identifer
+                              :details {:id    user-id
```

Comment:
These don't need to be passed explicitly. If you just pass in the entire user object through to `audit-log/record-event!` as the second arg, then it'll set `user_id` to `*current-user-id*` automatically, and `model_id` to the ID of the user object automatically, which is what we want.

For this event, since we want `details` to be empty, you could implement `model-details` to return `nil` for the `password-reset-initiated` event.